tests/lib/Controller/ApiControllerTest.php

```diff
@@ -1,137 +0,0 @@
-
-	/** @var  ShareStatistics | \PHPUnit_Framework_MockObject_MockObject */
-	private $shareStatistics;
-
-	/** @var  SessionStatistics | \PHPUnit_Framework_MockObject_MockObject */
-	private $sessionStatistics;
-
-	/** @var  ApiController */
-	private $instance;
-
-	public function setUp() {
-		parent::setUp();
-
-		$this->request = $this->getMockBuilder('OCP\IRequest')
-			->disableOriginalConstructor()
-			->getMock();
-
-		$this->systemStatistics = $this->getMockBuilder('OCA\ServerInfo\SystemStatistics')
-			->disableOriginalConstructor()
-			->getMock();
-
-		$this->storageStatistics = $this->getMockBuilder('OCA\ServerInfo\StorageStatistics')
-			->disableOriginalConstructor()
-			->getMock();
-
-		$this->phpStatistics = $this->getMockBuilder('OCA\ServerInfo\PhpStatistics')
-			->disableOriginalConstructor()
-			->getMock();
-
-		$this->databaseStatistics = $this->getMockBuilder('OCA\ServerInfo\DatabaseStatistics')
-			->disableOriginalConstructor()
-			->getMock();
-
-		$this->shareStatistics = $this->getMockBuilder('OCA\ServerInfo\ShareStatistics')
-			->disableOriginalConstructor()
-			->getMock();
-
-		$this->sessionStatistics = $this->getMockBuilder('OCA\ServerInfo\SessionStatistics')
-			->disableOriginalConstructor()
-			->getMock();
-
-
-		$this->instance = new ApiController(
-			'ServerInfoTest',
-			$this->request,
-			$this->systemStatistics,
-			$this->storageStatistics,
-			$this->phpStatistics,
-			$this->databaseStatistics,
-			$this->shareStatistics,
-			$this->sessionStatistics
-		);
-	}
-
-	public function testInfo() {
-
-		$this->systemStatistics->expects($this->once())->method('getSystemStatistics')
-			->willReturn('systemStatistics');
-		$this->storageStatistics->expects($this->once())->method('getStorageStatistics')
-			->willReturn('storageStatistics');
-		$this->phpStatistics->expects($this->once())->method('getPhpStatistics')
-			->willReturn('phpStatistics');
-		$this->databaseStatistics->expects($this->once())->method('getDatabaseStatistics')
-			->willReturn('databaseStatistics');
-		$this->shareStatistics->expects($this->once())->method('getShareStatistics')
-			->willReturn('shareStatistics');
-		$this->sessionStatistics->expects($this->once())->method('getSessionStatistics')
-			->willReturn('sessionStatistics');
-
-		$result = $this->instance->info();
-		$this->assertTrue($result instanceof DataResponse);
-		$data = $result->getData();
-		$this->assertTrue(isset($data['nextcloud']));
-		$this->assertTrue(isset($data['server']));
-
-		$this->assertSame('systemStatistics', $data['nextcloud']['system']);
-		$this->assertSame('storageStatistics', $data['nextcloud']['storage']);
-		$this->assertSame('shareStatistics', $data['nextcloud']['shares']);
-		$this->assertSame('unknown', $data['server']['webserver']);
-		$this->assertSame('databaseStatistics', $data['server']['database']);
-		$this->assertSame('phpStatistics', $data['server']['php']);
-		$this->assertSame('sessionStatistics', $data['activeUsers']);
```

Comment:
Maybe extend the tests instead of removing them. Then you can see when the API changes.